circle.yml

```diff
@@ -6,13 +6,7 @@ test:
     - NODE_ENV=test make client/config/index.js:
         parallel: true
   override:
-    - bin/run-lint :
-        parallel: true
-        files:
-          - client/**/*.js
-          - client/**/*.jsx
-          - server/**/*.js
-          - server/**/*.jsx
+    - bin/run-lint $(git diff --name-only $(git merge-base $(git rev-parse --abbrev-ref HEAD) origin/master)..HEAD *js *jsx)
```

Comment:
Exactly, that's the purpose of this PR.

> We still want to show problems in other files/lines, but just relegate errors to warnings.

I am not sure how to integrate this idea at a CircleCI level.

#### Rationale for the approach I have chosen with this PR

The idea makes a lot of sense in our development machines, as it gives us flexibility: you can solve `warnings` on areas aside of you task or you can focus of sending good code that passes validation without being prevented to move forward due to problems in other areas. That's the developer's call (taking into account the time you have, urgency of the task, etc).

But, as far as I have understood how CircleCI is being used now in calypso, it gives us a binary answer: _does the PR pass the quality checks or not? **Is it red or green?**_. Facing that choice, I asked myself: _should CircleCI fail or should pass for `warnings` in files the PR has not modified?_ I have came to the conclusion that it should not fail for the `warnings` in those files, as the reason we have this task in the first place is being able to move forward and produce new quality code meanwhile we migrate the old code to current practices. The current [run-lint script](https://github.com/Automattic/wp-calypso/blob/master/bin/run-lint#L4) adopts the same position by reporting only `errors` to CircleCI.

#### Conclusion

So: if we are going to report only as `errors` the rules we break in the files modified in the PR **and** only those `errors` would make CircleCI to fail, I think passing `run-lint` only the files modified in the PR is a sensible thing to do. From a CircleCI point of view, we do not need to compute the files we are not going to consider as errors. For free, we also have a potential gain in performance in doing so. 

Am I missing something?